scipy/spatial/transform/rotation.py

```diff
@@ -419,3 +507,55 @@ def from_euler(cls, seq, angles, degrees=False):
 
         quat = _elementary_quat_compose(seq, angles, intrinsic)
         return cls(quat[0] if is_single else quat, normalized=True)
+
+    def as_euler(self, seq, degrees=False):
+        """Return the Euler angles representation of the Rotation.
+
+        This function returns a numpy.ndarray of shape (N, 3) or (3,) depending
+        on how the object was initialized. The algorithm presented in [2]_ has
+        been adapted for our use to extract Euler angles, keeping in mind the
+        conventions presented in [3]_.
+
+        Parameters
+        ----------
+        seq : string, length 3
+            3 characters belonging to the set {'X', 'Y', 'Z'} for intrinsic
+            rotations, or {'x', 'y', 'z'} for extrinsic rotations [1]_.
+            Adjacent axes cannot be the same.
+            Extrinsic and intrinsic rotations cannot be mixed in one function
+            call.
+
+        degrees : boolean, optional
+            Returned angles are in degrees if this flag is True, else they are
+            in radians. Default is False.
+
+        References
+        ----------
+        .. [1] `Euler angle definitions
+                <https://en.wikipedia.org/wiki/Euler_angles#Definition_by_intrinsic_rotations>`_
+        .. [2] Malcolm D. Shuster, F. Landis Markley
+                `General Formula for Euler Angles
+                <https://arc.aiaa.org/doi/abs/10.2514/1.16622>`_
+        .. [3] Malcolm D. Shuster `A Survey of Attitude Representations
```

Comment:
To be honest I'm not sure what reference [3] has to do with the conventions we use. In fact he uses the opposite convention (which I really don't like by the way).

Reply:
So I don't think I was clear here. I meant to say that the conventions used here _are_ in fact the opposite of what we use. I included it only if someone was going though reference 2 and wanted to know about the implementation.

Should I make this difference in conventions clearer or remove reference 3 completely?